Fix image-upload-non-auth.js

Blocks #2106.

```
✓ firefox on darwin 17.4.0 - Image Uploads (non-auth) - postMessaged dataURI preview (2.17s)
✓ firefox on darwin 17.4.0 - Image Uploads (non-auth) - postMessaged blob preview (6.5s)
✓ firefox on darwin 17.4.0 - Image Uploads (non-auth) - uploaded image file preview (1.596s)
✓ firefox on darwin 17.4.0 - Image Uploads (non-auth) - postMessaged dataURI image doesn't upload before form submission (1.765s)
✓ firefox on darwin 17.4.0 - Image Uploads (non-auth) - postMessaged blob image doesn't upload before form submission (1.527s)
✓ firefox on darwin 17.4.0 - Image Uploads (non-auth) - uploaded image file doesn't upload before form submission (1.63s)
Error fetching http://localhost:5000/?open=1
× firefox on darwin 17.4.0 - Image Uploads (non-auth) - remove image upload button (10.407s)
    NoSuchElement: [POST http://localhost:4444/wd/hub/session/84faf848-424a-9049-82af-a7766c86da5d/element / {"using":"css selector","value":".remove-upload"}] Unable to locate element: .remove-upload
    For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
    Build info: version: '3.5.2', revision: '10229a9020', time: '2017-08-21T17:54:21.164Z'
    System info: host: 'omg-5', ip: '192.168.2.83', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.13.3', java.version: '1.8.0_144'
    Driver info: driver.version: unknown
      at Object.openPage  <tests/functional/lib/helpers.js:27:8>
      at Test.remove image upload button [as test]  <tests/functional/image-uploads-non-auth.js:138:27>
      at <src/lib/Test.ts:260:47>
      at Command.<anonymous>  <tests/functional/lib/helpers.js:36:12>
      at Object.openPage  <tests/functional/lib/helpers.js:29:8>
      at Test.remove image upload button [as test]  <tests/functional/image-uploads-non-auth.js:138:27>
      at <src/lib/Test.ts:260:47>

Error fetching http://localhost:5000/?open=1
× firefox on darwin 17.4.0 - Image Uploads (non-auth) - double image select works (10.404s)
    NoSuchElement: [POST http://localhost:4444/wd/hub/session/84faf848-424a-9049-82af-a7766c86da5d/element / {"using":"css selector","value":".remove-upload"}] Unable to locate element: .remove-upload
    For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
    Build info: version: '3.5.2', revision: '10229a9020', time: '2017-08-21T17:54:21.164Z'
    System info: host: 'omg-5', ip: '192.168.2.83', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '10.13.3', java.version: '1.8.0_144'
    Driver info: driver.version: unknown
      at Object.openPage  <tests/functional/lib/helpers.js:27:8>
      at Test.double image select works [as test]  <tests/functional/image-uploads-non-auth.js:178:32>
      at <src/lib/Test.ts:260:47>
      at Command.<anonymous>  <tests/functional/lib/helpers.js:36:12>
      at Object.openPage  <tests/functional/lib/helpers.js:29:8>
      at Test.double image select works [as test]  <tests/functional/image-uploads-non-auth.js:178:32>
      at <src/lib/Test.ts:260:47>
```

@miketaylr What I have done:

https://gist.github.com/magsout/3d2cb9d61feea65367039c5afb0b5258/revisions

But I don't understand the last error:

```
× firefox on darwin 17.3.0 - Image Uploads (non-auth) - remove image upload button
AssertionError: Remove button is displayed: expected false to equal true
```

For me the selector is ok and should be working

```
.findByCssSelector(".js-image-upload .js-remove-upload")
          .isDisplayed()
          .then(function(isDisplayed) {
            assert.equal(isDisplayed, true, "Remove button is displayed");
          })
          .end()
```

thought?

--- a/tests/functional/comments-auth.js
+++ b/tests/functional/comments-auth.js
@@ -62,27 +62,26 @@ registerSuite("Comments (auth)", {
     "Posting an empty comment fails"() {
       var originalCommentsLength;
       var allCommentsLength;
-      return (
-        FunctionalHelpers.openPage(this, url("/issues/100"), ".js-Issue")
-          .findAllByCssSelector(".js-Issue-comment")
-          .then(function(elms) {
-            originalCommentsLength = elms.length;
-          })
-          .end()
+      return FunctionalHelpers.openPage(this, url("/issues/100"), ".js-Issue")
+        .findAllByCssSelector(".js-Issue-comment")
+        .then(function(elms) {
+          originalCommentsLength = elms.length;
+        })
+        .end()
+        .execute(() => {
           // click the comment button
-          .findByCssSelector(".js-Issue-comment-button")
-          .click()
-          .end()
-          .sleep(2000)
-          .findAllByCssSelector(".js-Issue-comment")
-          .then(function(elms) {
-            allCommentsLength = elms.length;
-            assert(
-              originalCommentsLength === allCommentsLength,
-              "Comment was not successfully left."
-            );
-          })
-      );
+          $(".js-Issue-comment-button")[0].click();
+        })
+        .end()
+        .sleep(2000)
+        .findAllByCssSelector(".js-Issue-comment")
+        .then(function(elms) {
+          allCommentsLength = elms.length;
+          assert(
+            originalCommentsLength === allCommentsLength,
+            "Comment was not successfully left."
+          );
+        });
     },
 
     "Add a screenshot to a comment"() {
@@ -114,7 +113,6 @@ registerSuite("Comments (auth)", {
         ".js-Comment-text"
       )
         .findByCssSelector(".js-Comment-text")
-        .click()
         .type("g")
         .end()
         .setFindTimeout(2000)
@@ -132,7 +130,6 @@ registerSuite("Comments (auth)", {
         ".js-Comment-text"
       )
         .findByCssSelector(".js-Comment-text")
-        .click()
         .type("l")
         .end()
         .setFindTimeout(2000)

--- a/tests/functional/image-uploads-non-auth.js
+++ b/tests/functional/image-uploads-non-auth.js
@@ -8,7 +8,12 @@ const { assert } = intern.getPlugin("chai");
 const { registerSuite } = intern.getInterface("object");
 const FunctionalHelpers = require("./lib/helpers.js");
 
-var url = intern.config.siteRoot + "/?open=1";
+const url = intern.config.siteRoot + "/?open=1";
+
+// This string is executed by calls to `execute()` in various tests
+// it postMessages a small green test square.
+const POSTMESSAGE_TEST_SQUARE =
+  'postMessage("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABkAAAAZCAIAAABLixI0AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAAB3RJTUUH3gYSAig452t/EQAAAClJREFUOMvtzkENAAAMg0A25ZU+E032AQEXoNcApCGFLX5paWlpaWl9dqq9AS6CKROfAAAAAElFTkSuQmCC", "http://localhost:5000")';
 
 registerSuite("Image Uploads (non-auth)", {
   tests: {
@@ -16,9 +21,7 @@ registerSuite("Image Uploads (non-auth)", {
       return (
         FunctionalHelpers.openPage(this, url, ".js-image-upload")
           // send a small base64 encoded green test square
-          .execute(
-            'postMessage("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABkAAAAZCAIAAABLixI0AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAAB3RJTUUH3gYSAig452t/EQAAAClJREFUOMvtzkENAAAMg0A25ZU+E032AQEXoNcApCGFLX5paWlpaWl9dqq9AS6CKROfAAAAAElFTkSuQmCC", "http://localhost:5000")'
-          )
+          .execute(POSTMESSAGE_TEST_SQUARE)
           .sleep(1000)
           .findByCssSelector(".js-image-upload")
           .getAttribute("style")
@@ -77,9 +80,7 @@ registerSuite("Image Uploads (non-auth)", {
       return (
         FunctionalHelpers.openPage(this, url, ".js-image-upload")
           // send a small base64 encoded green test square
-          .execute(
-            'postMessage("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABkAAAAZCAIAAABLixI0AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAAB3RJTUUH3gYSAig452t/EQAAAClJREFUOMvtzkENAAAMg0A25ZU+E032AQEXoNcApCGFLX5paWlpaWl9dqq9AS6CKROfAAAAAElFTkSuQmCC", "http://localhost:5000")'
-          )
+          .execute(POSTMESSAGE_TEST_SQUARE)
           .sleep(1000)
           .findByCssSelector("#steps_reproduce")
           .getProperty("value")
@@ -137,9 +138,7 @@ registerSuite("Image Uploads (non-auth)", {
       return (
         FunctionalHelpers.openPage(this, url, ".remove-upload")
           // send a small base64 encoded green test square
-          .execute(
-            'postMessage("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABkAAAAZCAIAAABLixI0AAAACXBIWXMAAA7DAAAOwwHHb6hkAAAAB3RJTUUH3gYSAig452t/EQAAAClJREFUOMvtzkENAAAMg0A25ZU+E032AQEXoNcApCGFLX5paWlpaWl9dqq9AS6CKROfAAAAAElFTkSuQmCC", "http://localhost:5000")'
-          )
+          .execute(POSTMESSAGE_TEST_SQUARE)
           .sleep(1000)
           .findByCssSelector(".js-image-upload .remove-upload")
           .isDisplayed()

--- a/tests/functional/image-uploads-non-auth.js
+++ b/tests/functional/image-uploads-non-auth.js
@@ -7,8 +7,9 @@ const intern = require("intern").default;
 const { assert } = intern.getPlugin("chai");
 const { registerSuite } = intern.getInterface("object");
 const FunctionalHelpers = require("./lib/helpers.js");
+const keys = require("@theintern/leadfoot/keys").default;
 
-const url = intern.config.siteRoot + "/?open=1";
+const url = intern.config.siteRoot + "/issues/new";
 
 // This string is executed by calls to `execute()` in various tests
 // it postMessages a small green test square.
@@ -62,6 +63,7 @@ registerSuite("Image Uploads (non-auth)", {
       return FunctionalHelpers.openPage(this, url, ".js-image-upload")
         .findById("image")
         .type("tests/fixtures/green_square.png")
+        .pressKeys(keys.RETURN)
         .sleep(1000)
         .end()
         .findByCssSelector(".js-image-upload")
@@ -120,7 +122,7 @@ registerSuite("Image Uploads (non-auth)", {
       return FunctionalHelpers.openPage(this, url, ".js-image-upload")
         .findById("image")
         .type("tests/fixtures/green_square.png")
-        .sleep(1000)
+        .pressKeys(keys.RETURN)
         .end()
         .findByCssSelector("#steps_reproduce")
         .getProperty("value")
@@ -134,21 +136,32 @@ registerSuite("Image Uploads (non-auth)", {
         .end();
     },
 
-    "remove image upload button"() {
+    "remove image upload button is shown"() {
       return (
-        FunctionalHelpers.openPage(this, url, ".remove-upload")
+        FunctionalHelpers.openPage(this, url, ".js-remove-upload")
           // send a small base64 encoded green test square
           .execute(POSTMESSAGE_TEST_SQUARE)
           .sleep(1000)
-          .findByCssSelector(".js-image-upload .remove-upload")
+          .findByCssSelector(".js-remove-upload")
           .isDisplayed()
           .then(function(isDisplayed) {
             assert.equal(isDisplayed, true, "Remove button is displayed");
           })
           .end()
-          .findByCssSelector(".js-image-upload .remove-upload")
-          .click()
+      );
+    },
+
+    "clicking remove image upload button removes preview"() {
+      return (
+        FunctionalHelpers.openPage(this, url, ".js-remove-upload")
+          // send a small base64 encoded green test square
+          .execute(POSTMESSAGE_TEST_SQUARE)
           .sleep(1000)
+          .execute(() => {
+            // work around for chrome "Other element would receive the click"
+            // error
+            $(".js-remove-upload")[0].click();
+          })
           .end()
           .findByCssSelector(".js-image-upload")
           .getAttribute("style")
@@ -160,13 +173,28 @@ registerSuite("Image Uploads (non-auth)", {
             );
           })
           .end()
-          .findByCssSelector("#steps_reproduce")
-          .getProperty("value")
-          .then(function(val) {
+      );
+    },
+
+    "clicking remove image upload button removes image URL"() {
+      return (
+        FunctionalHelpers.openPage(this, url, ".js-remove-upload")
+          // send a small base64 encoded green test square
+          .execute(POSTMESSAGE_TEST_SQUARE)
+          .sleep(1000)
+          .execute(() => {
+            // work around for chrome "Other element would receive the click"
+            // error
+            $(".js-remove-upload")[0].click();
+          })
+          .end()
+          .findByCssSelector(".js-image-upload")
+          .getAttribute("style")
+          .then(function(inlineStyle) {
             assert.notInclude(
-              val,
-              "[![Screenshot Description](http://localhost:5000/uploads/",
-              "The url to the image upload was correctly removed."
+              inlineStyle,
+              "data:image/png;base64,iVBOR",
+              "Preview was removed"
             );
           })
           .end()
@@ -174,19 +202,20 @@ registerSuite("Image Uploads (non-auth)", {
     },
 
     "double image select works"() {
-      return FunctionalHelpers.openPage(this, url, ".remove-upload")
+      return FunctionalHelpers.openPage(this, url, ".js-remove-upload")
         .findById("image")
         .type("tests/fixtures/green_square.png")
+        .pressKeys(keys.RETURN)
         .sleep(1000)
         .end()
-        .findByCssSelector(".js-image-upload .remove-upload")
+        .findByCssSelector(".js-remove-upload")
         .click()
-        .sleep(1000)
         .end()
         .findById("image")
         .type("tests/fixtures/green_square.png")
-        .end()
+        .pressKeys(keys.RETURN)
         .sleep(1000)
+        .end()
         .findByCssSelector(".js-image-upload")
         .getAttribute("style")
         .then(function(inlineStyle) {

--- a/tests/functional/image-uploads-non-auth.js
+++ b/tests/functional/image-uploads-non-auth.js
@@ -7,7 +7,6 @@ const intern = require("intern").default;
 const { assert } = intern.getPlugin("chai");
 const { registerSuite } = intern.getInterface("object");
 const FunctionalHelpers = require("./lib/helpers.js");
-const keys = require("@theintern/leadfoot/keys").default;
 
 const url = intern.config.siteRoot + "/issues/new";
 
@@ -36,7 +35,6 @@ registerSuite("Image Uploads (non-auth)", {
           .end()
       );
     },
-
     "postMessaged blob preview"() {
       return (
         FunctionalHelpers.openPage(this, url, ".js-image-upload")
@@ -63,9 +61,8 @@ registerSuite("Image Uploads (non-auth)", {
       return FunctionalHelpers.openPage(this, url, ".js-image-upload")
         .findById("image")
         .type("tests/fixtures/green_square.png")
-        .pressKeys(keys.RETURN)
-        .sleep(1000)
         .end()
+        .sleep(1000)
         .findByCssSelector(".js-image-upload")
         .getAttribute("style")
         .then(function(inlineStyle) {
@@ -122,8 +119,8 @@ registerSuite("Image Uploads (non-auth)", {
       return FunctionalHelpers.openPage(this, url, ".js-image-upload")
         .findById("image")
         .type("tests/fixtures/green_square.png")
-        .pressKeys(keys.RETURN)
         .end()
+        .sleep(1000)
         .findByCssSelector("#steps_reproduce")
         .getProperty("value")
         .then(function(val) {
@@ -203,19 +200,17 @@ registerSuite("Image Uploads (non-auth)", {
 
     "double image select works"() {
       return FunctionalHelpers.openPage(this, url, ".js-remove-upload")
-        .findById("image")
-        .type("tests/fixtures/green_square.png")
-        .pressKeys(keys.RETURN)
+        .execute(POSTMESSAGE_TEST_SQUARE)
         .sleep(1000)
-        .end()
-        .findByCssSelector(".js-remove-upload")
-        .click()
-        .end()
-        .findById("image")
-        .type("tests/fixtures/green_square.png")
-        .pressKeys(keys.RETURN)
+        .execute(() => {
+          $(".js-image-upload")[0].click();
+        })
+        .execute(POSTMESSAGE_TEST_SQUARE)
         .sleep(1000)
-        .end()
+        .execute(() => {
+          $(".js-image-upload")[0].click();
+        })
+        .sleep(1000)
         .findByCssSelector(".js-image-upload")
         .getAttribute("style")
         .then(function(inlineStyle) {

--- a/tests/functional/labels-auth.js
+++ b/tests/functional/labels-auth.js
@@ -33,6 +33,9 @@ registerSuite("Labels (auth)", {
         .end()
         .findByCssSelector(".js-CategoryEditor")
         .end()
+        .findByCssSelector(".js-CategoryEditor-close")
+        .click()
+        .end()
         .findByCssSelector(".js-LabelEditorLauncher")
         .click()
         .getAttribute("class")
@@ -76,7 +79,7 @@ registerSuite("Labels (auth)", {
         .end()
         .findByCssSelector(".js-CategoryEditor")
         .end()
-        .findByCssSelector("main")
+        .findByCssSelector("footer")
         .click()
         .end()
         .findByCssSelector(".js-LabelEditorLauncher")
@@ -87,16 +90,16 @@ registerSuite("Labels (auth)", {
         .end();
     },
 
-    "Clicking close button actually closes it?": function() {
+    "Clicking close button actually closes it": function() {
       return FunctionalHelpers.openPage(
         this,
         url("/issues/2"),
         ".js-LabelEditorLauncher",
         true /* longerTimeout */
       )
-        .findByCssSelector(".js-LabelEditorLauncher")
-        .click()
-        .end()
+        .execute(() => {
+          $(".js-LabelEditorLauncher")[0].click();
+        })
         .findByCssSelector(".js-CategoryEditor-close")
         .click()
         .end()

--- a/tests/functional/milestones-auth.js
+++ b/tests/functional/milestones-auth.js
@@ -33,6 +33,9 @@ registerSuite("Milestones (auth)", {
         .end()
         .findByCssSelector(".js-CategoryEditor")
         .end()
+        .findByCssSelector(".js-CategoryEditor-close")
+        .click()
+        .end()
         .findByCssSelector(".js-MilestoneEditorLauncher")
         .click()
         .getAttribute("class")
@@ -76,7 +79,7 @@ registerSuite("Milestones (auth)", {
         .end()
         .findByCssSelector(".js-CategoryEditor")
         .end()
-        .findByCssSelector("main")
+        .findByCssSelector("footer")
         .click()
         .end()
         .findByCssSelector(".js-MilestoneEditorLauncher")
@@ -87,16 +90,16 @@ registerSuite("Milestones (auth)", {
         .end();
     },
 
-    "Clicking close button actually closes it?"() {
+    "Clicking close button actually closes it"() {
       return FunctionalHelpers.openPage(
         this,
         url("/issues/2"),
         ".js-MilestoneEditorLauncher",
         true /* longerTimeout */
       )
-        .findByCssSelector(".js-MilestoneEditorLauncher")
-        .click()
-        .end()
+        .execute(() => {
+          $(".js-MilestoneEditorLauncher")[0].click();
+        })
         .findByCssSelector(".js-CategoryEditor-close")
         .click()
         .end()

--- a/webcompat/static/css/lib/reset-add.css
+++ b/webcompat/static/css/lib/reset-add.css
@@ -22,14 +22,19 @@ svg {
   fill: var(--color-second);
 }
 
+/* Prevent svg elements inside buttons from swallowing events */
+button svg {
+  pointer-events: none;
+}
+
 /* Sets Boxmodel overall the project */
-*, 
+*,
 *::after,
 *::before {
   box-sizing: border-box;
 }
 
 ol,
-ol li{
+ol li {
   list-style: decimal;
 }

--- a/webcompat/static/css/src/bio.css
+++ b/webcompat/static/css/src/bio.css
@@ -21,11 +21,13 @@
   margin-top: 20px;
 }
 
-.bio .grid-nested:first-child, .grid-cell {
+.bio .grid-nested:first-child,
+.grid-cell {
   margin-right: 10px;
 }
 
-.bio .grid-nested:last-child, .grid-cell.bio-desc {
+.bio .grid-nested:last-child,
+.grid-cell.bio-desc {
   padding-top: 0;
   margin-top: -5px;
 }

--- a/webcompat/static/css/src/form.css
+++ b/webcompat/static/css/src/form.css
@@ -222,10 +222,6 @@
   visibility: hidden;
 }
 
-.label-remove {
-  visibility: hidden;
-}
-
 .is-validated .label-remove {
   visibility: visible;
   z-index: 40;

--- a/webcompat/static/css/src/issues-list.css
+++ b/webcompat/static/css/src/issues-list.css
@@ -88,7 +88,6 @@
   }
 }
 
-
 #x-search-bar .nav-button {
   margin-top: -23px;
   position: absolute;

